File: 0070-climbing-stairs/solution.c
int climbStairs(int n) {
    int sum = 0;
    int a = 0;
    int b = 1;
    int c = 0;
    int i = 0;
    while(i<n){
        c = a + b;
        a = b;
        b = c;
        i ++;
    }
    return c;
}

File: 0070-climbing-stairs/solution.java
class Solution {
    HashMap<Integer, Integer> map = new HashMap<>();

    public int climbStairs(int n) {
        if (n == 0) {
            return 1;
        }

        if (n < 0) {
            return 0;
        }

        if (map.containsKey(n)) {
            return map.get(n);
        }

        int val = climbStairs(n - 1) + climbStairs(n - 2);
        map.put(n, val);
        return val;
    }
}
